client/src/hooks/tasks/index.ts

```diff
@@ -100,3 +100,27 @@ export function useUpdateTask(): UseMutationResult<TaskAPIResponse> {
     },
   );
 }
+
+export function useTaskErrors(
+  id: Task['id'],
+  options: UseQueryOptions<string> = {},
+): UseQueryResult<string> {
+  const query = useQuery<string>(
+    ['task-error', id],
+    () =>
+      apiRawService
+        .request({
+          method: 'GET',
+          url: `/tasks/report/errors/${id}`,
+          params: {
+            type: 'sourcing_data_import',
+          },
+        })
+        .then(({ data: responseData }) => responseData),
+    {
+      ...DEFAULT_QUERY_OPTIONS,
+      ...options,
+    },
+  );
+  return query;
+}
```

Comment:
```suggestion
export function useDownloadTaskErrors(
  id: Task['id'],
  options: UseQueryOptions<string> = {},
): UseQueryResult<string> {
  const query = useQuery<string>(
    ['task-error', id],
    () =>
      apiRawService
        .request({
          method: 'GET',
          url: `/tasks/report/errors/${id}`,
          params: {
            type: 'sourcing_data_import',
          },
        })
        .then(response => response.blob())
        .then(blob => {
            var url = window.URL.createObjectURL(blob);
            var a = document.createElement('a');
            a.href = url;
            a.download = "filename.xlsx";
            document.body.appendChild(a); // we need to append the element to the dom -> otherwise it will not work in firefox
            a.click();    
            a.remove();  //afterwards we remove the element again         
        }),
    {
      ...DEFAULT_QUERY_OPTIONS,
      ...options,
    },
  );
  return query;
}
```

This is another way to make it downloadable. I like this way because it doesn't parse the data, it just downloads from API. So, it has better performance.

Reply:
It didn’t work well for me, so I am using the  previous way